tests/test_integration.py

```diff
@@ -0,0 +1,801 @@
+import logging
+import mock
+import pytest
+import requests_mock
+from copy import deepcopy
+
+from pubtools._quay.push_docker import PushDocker
+from pubtools._quay.tag_docker import TagDocker
+from pubtools._quay import iib_operations
+from .utils.misc import sort_dictionary_sortable_values, compare_logs
+
+# flake8: noqa: E501
+
+
+@mock.patch("pubtools._quay.operator_pusher.run_entrypoint")
+@mock.patch("pubtools._quay.tag_images.send_umb_message")
+@mock.patch("pubtools._quay.command_executor.RemoteExecutor._run_cmd")
+@mock.patch("pubtools._quay.signature_handler.proton")
+@mock.patch("pubtools._quay.signature_handler.ManifestClaimsHandler")
+@mock.patch("pubtools._quay.signature_handler.run_entrypoint")
+@mock.patch("pubtools._quay.push_docker.run_entrypoint")
+def test_push_docker_multiarch_merge_ml_operator(
+    mock_run_entrypoint_push_docker,
+    mock_run_entrypoint_sig_handler,
+    mock_claims_handler,
+    mock_proton,
+    mock_run_cmd,
+    mock_send_umb_message,
+    mock_run_entrypoint_operator_pusher,
+    target_settings,
+    container_multiarch_push_item_integration,
+    operator_push_item_ok,
+    src_manifest_list,
+    dest_manifest_list,
+):
+    class IIBRes:
+        def __init__(self, index_image, index_image_resolved):
+            self.index_image = index_image
+            self.index_image_resolved = index_image_resolved
+
+    # hub usage has to be mocked
+    hub = mock.MagicMock()
+    mock_get_target_info = mock.MagicMock()
+    mock_get_target_info.return_value = {"settings": {"quay_namespace": "stage-namespace"}}
+    hub.worker.get_target_info = mock_get_target_info
+    target_settings["propagated_from"] = "test-target"
+
+    mock_run_entrypoint_push_docker.side_effect = [
+        # pubtools-pyxis-get-repo-metadata
+        {"release_categories": ["definitely-not-deprecated"]},
+    ]
+    mock_run_entrypoint_sig_handler.side_effect = [
```

Comment:
If we wanted to add calls or change call parameters we would still have to revisit this. It would probably be more laborious. I suggest we leave it as-is. What do you think?